### PR TITLE
Update libp2p, async-std, and other deps

### PR DESCRIPTION
**Pull request by austinabell**
**Summary of changes**
Changes introduced in this pull request:
- Migrates to new error prone async-std channels (0.1.8)
- Libp2p update to 0.28
  - Updated bitswap dep and reinitialized request-response (I kept their oneshot channels)

**Reference issue to close (if applicable)**

Closes 

**Other information and links**

### CODE REVIEW

**Comment by austinabell** [comment]
Having issues with libp2p connections timing out with this new version, going to let this sit for a bit to debug. Could also just remove that version bump from this PR and revisit later, undecided.

**Comment by ec2** [comment]
Interesting... When does it actually time out? When its trying to connect? Or when you try to send some RPC(BlockSync) req?

**Comment by austinabell** [comment]
> Interesting... When does it actually time out? When its trying to connect? Or when you try to send some RPC(BlockSync) req?

When dialing trying to send rpc requests

**Comment by austinabell** [comment]
I think the issue is actually because https://github.com/libp2p/rust-libp2p/pull/1867 dropped support for one-way requests (we are not responding to Hello currently) and possibly this is blocking us from being able to send chain exchange requests to that same peer.

I'm going to investigate a bit more but I'm just leaning toward updating our hello response (was going to be done later anyway) and following this pattern, so we are available for future updates of the upstream request-response

**Comment by austinabell** [comment]
Going to just leave it as 0.28 update for now and open a draft PR for 0.34 changes (it's currently broken running against the go implementation)

File: node/forest_libp2p/src/service.rs [hunk 1]
```diff
@@ -368,32 +348,41 @@ where
...
+            .into_authentic(&local_key)
+            .expect("Noise key generation failed");
+
+        noise::NoiseConfig::xx(dh_keys).into_authenticated()
+    };
+
+    let mplex_config = {
+        let mut mplex_config = mplex::MplexConfig::new();
+        mplex_config.max_buffer_len(usize::MAX);
+        // mplex_config.set_max_buffer_size(usize::MAX);
```
Comment (ec2) [inline node/forest_libp2p/src/service.rs:373]: Did you forget to remove this comment?

File: node/forest_libp2p/src/behaviour.rs [hunk 2]
```diff
@@ -55,6 +59,18 @@ pub struct ForestBehaviour {
     events: Vec<ForestBehaviourEvent>,
     #[behaviour(ignore)]
     peers: HashSet<PeerId>,
+    /// Keeps track of Chain exchange requests to responses
+    #[behaviour(ignore)]
+    cx_request_table: HashMap<RequestId, OneShotSender<ChainExchangeResponse>>,
+    /// Boxed futures of responses for the
```
Comment (ec2) [inline node/forest_libp2p/src/behaviour.rs:65]: for the.....

File: node/forest_libp2p/src/service.rs (hunk 1 above)
Comment (austinabell) [reply node/forest_libp2p/src/service.rs:373]: ah ya, it was what the type changed in future libp2p version, I commented for easier switching, but #928 exists

File: node/forest_libp2p/src/behaviour.rs (hunk 2 above)
Comment (austinabell) [reply node/forest_libp2p/src/behaviour.rs:65]: haha

**Review by ec2 — APPROVED** [review]

**Pull request merged**